documentloader/documentloader.go

```diff
@@ -0,0 +1,14 @@
+package documentloader
+
+import (
+	"github.com/tmc/langchaingo/schema"
+	"github.com/tmc/langchaingo/textsplitter"
+)
+
+// Loader is the interface for loading and splitting documents from a source.
+type Loader interface {
+	// Loads loads from a source and returns documents.
+	Load() ([]schema.Document, error)
```

Comment:
since document loader /could/ involve some network or other io we probably could accept a context.Context in this interface -- thoughts?